lib/model/port.js

```diff
@@ -0,0 +1,79 @@
+/**
+ * port.js
+ *
+ * Definition for the Port class. Ports are the points on a Node where
+ * connections attach to or from the node. They can be either inputs
+ * or outputs.
+ */
+
+/**
+ * Constructor function.
+ *
+ * @param id - The Port's unique identifier.
+ * @param type - The type of port. Valid options include "input" or "output".
+ */
+var Port = function(id,type) {
+	this.id = id || 0;
+	this.x = 0;

```

Comment:
Remember what we talked on tuesday? Its kinda hard to explain without a picture but its the bottom left edge of the node is (0,0) the top right edge is (100,100), using the edges of the node to put the points down. 
![image](https://cloud.githubusercontent.com/assets/1589075/5993196/958c5bc8-aa15-11e4-9090-13565d28445a.png)